File: 2655-rearrange-array-to-maximize-prefix-score/rearrange-array-to-maximize-prefix-score.cpp
class Solution {
public:
    int maxScore(vector<int>& nums) {
        //they are asking after making some prefix sum array give the number of positive
        ///numbers in it but that should be maximized that's all the question is

        //simple greedy approach is enough
        sort(nums.begin(), nums.end(), greater<int>());
        // if (nums[0] <= 0) return 0; //first element itsel is negative means nothing pos is possible

        long long ans = 0;
        int i = 0;

        for (i = 0; i < nums.size(); i++) {
            ans += nums[i];
            if (ans <= 0) break;
        }

        return i;
    }
};

File: 2655-rearrange-array-to-maximize-prefix-score/rearrange-array-to-maximize-prefix-score.java
class Solution {
    public int maxScore(int[] nums) {
        long sum = 0;
        int n = nums.length;
        int i = 0;
        Arrays.sort(nums);

        for (int j = 0; j < n / 2; j++) {
            int temp = nums[j];
            nums[j] = nums[n - 1 - j];
            nums[n - 1 - j] = temp;
        }

        while ( i < n ) {
            sum += nums[i];
            if (sum <= 0) return i;
            i++;
        }

        return i;
    }
}